Fix for Microblaze gcc misrendering long hex constants [gcc bz 65649]

--- gcc/config/microblaze/microblaze.c	(revision 222257)
+++ gcc/config/microblaze/microblaze.c	(working copy)
@@ -1661,7 +1661,7 @@
 	{			/* Looking for major  */
           if (*p == '.')
             {
-              *v++;
+              v++;
             }
           else
             {
@@ -2404,7 +2404,7 @@
 	  REAL_VALUE_TYPE value;
 	  REAL_VALUE_FROM_CONST_DOUBLE (value, op);
 	  REAL_VALUE_TO_TARGET_SINGLE (value, value_long);
-	  fprintf (file, HOST_WIDE_INT_PRINT_HEX, value_long);
+	  fprintf (file, "0x%lx", value_long);
 	}
       else
 	{
@@ -2463,7 +2463,7 @@
       print_operand_address (file, XEXP (op, 0));
     }
   else if (letter == 'm')
-    fprintf (file, HOST_WIDE_INT_PRINT_DEC, (1L << INTVAL (op)));
+    fprintf (file, "%ld", (1L << INTVAL (op)));
   else
     output_addr_const (file, op);
 }
